key.ui/src/main/java/de/uka/ilkd/key/core/KeYMediator.java

```diff
@@ -103,6 +105,10 @@ public class KeYMediator {
      * Currently opened proofs.
      */
     private final DefaultListModel<Proof> currentlyOpenedProofs = new DefaultListModel<>();
+    /**
+     * Currently loaded proofs.
+     */
+    private final WeakHashMap<Proof, Object> loadedProofs = new WeakHashMap<>();
```

Comment:
As discussed in our meeting: Since this is just used as a set (the value is always set to null), it may be possible to just use the list model instead. If not, please state in the comment that it is used as a set here.

Reply:
I had another look. It seems the `currentlyOpenedProofs` aren't actually managed by the mediator, they are just stored there. It is difficult to figure out exactly when proofs are added to this list. For now I think it is easier to handle duplicated proof loaded events (by removing this map and associated checks).